oauthproxy.go

```diff
@@ -208,12 +315,16 @@ func NewOAuthProxy(opts *Options, validator func(string) bool) *OAuthProxy {
 			SignatureHeader, SignatureHeaders)
 	}
 	for _, u := range opts.proxyURLs {
-		path := u.Path
+		path := u.EscapedPath()
 		host := u.Host
 		switch u.Scheme {
 		case httpScheme, httpsScheme:
+			if u.Fragment != "" {
+				path = u.Fragment
+			}
 			logger.Printf("mapping path %q => upstream %q", path, u)
-			proxy := NewWebSocketOrRestReverseProxy(u, opts, auth)
+			proxy := NewWebSocketOrRestReverseProxy(u, path, opts, auth)
+			checkMuxPath(path)
```

Comment:
Was there a plan to do something with the return value of this method?

Reply:
Not in its current usage. I pulled the warning into the function itself and forgot to update the signature.

Reply:
Can you update the signature please